addons/website/models/website.py

```diff
@@ -872,40 +872,16 @@ def _handle_views_and_pages(views):
 
     def _get_alternate_languages(self, canonical_params):
         self.ensure_one()
-
-        if not self._is_canonical_url(canonical_params=canonical_params):
-            # no hreflang on non-canonical pages
-            return []
-
-        languages = self.language_ids
-        if len(languages) <= 1:
-            # no hreflang if no alternate language
-            return []
-
         langs = []
-        shorts = []
-
-        for lg in languages:
-            lg_codes = lg.code.split('_')
-            short = lg_codes[0]
-            shorts.append(short)
+        for lg in self.env['res.lang'].get_available():
             langs.append({
-                'hreflang': ('-'.join(lg_codes)).lower(),
-                'short': short,
+                'id': lg[5],
+                'code': lg[0],
+                'url_code': lg[1],
+                'name': lg[2].split('/').pop(),
+                'flag_image_url': lg[4],
                 'href': self._get_canonical_url_localized(lang=lg, canonical_params=canonical_params),
             })
-
-        # if there is only one region for a language, use only the language code
-        for lang in langs:
-            if shorts.count(lang['short']) == 1:
-                lang['hreflang'] = lang['short']
```

Comment:
lost/broken, see other comment more detailed about it

Reply:
thanks, I fixed this one.